### ISSUE TITLE
add field validation

### ISSUE THREAD

**Issue by OlivierJM** [SparkEdUAB/sparked-server#174]
Currently validation is only happening on the client side and as much as we are trying to not send any unclean data it is always a good practive to have validation on both side.
ensure that all data sent to the server is as expected.

**Comment by OlivierJM** [comment]
We could leverage GraphQL capabilities or use something like yup to validate.

**Comment by Joebbrien** [comment]
I think yup would be best and is easy to use, this should be quick if we know the fields that are coming in from the client.

**Comment by OlivierJM** [comment]
> I think yup would be best and is easy to use, this should be quick if we know the fields that are coming in from the client.

Yeah fortunately all the fields are set in advance so it is easy to know 
the first place to check would be models here
https://github.com/SparkEdUAB/sparked-server/blob/master/src/models/user.js

then also user Type or any other Type being used can tell what exactly those fields are.
https://github.com/SparkEdUAB/sparked-server/blob/7f266468c15308df6b860448ce9c720cd21838df/src/typdefs/user.js

**Comment by Joebbrien** [comment]
Once I figure out how Apollo and GraghQL works, I can get done with validation, I have seen you do not have a lot of tables

**Comment by OlivierJM** [comment]
Yeahh I only have one db and few collections that are related in some ways but not directly 

check a draft of the design doc here https://github.com/SparkEdUAB/sparked-spec this is still work in progress to include other details

**Comment by Joebbrien** [comment]
I have added yup for register/login mutations. You can check how they fare and let me know. Though the validation is not detailed

**Comment by Joebbrien** [comment]
repo https://github.com/Joebbrien/sparked-server

**Comment by Joebbrien** [comment]
if that works, we can do the same with creation of courses/topics as well as queuing and updating

**Comment by OlivierJM** [comment]
I have checked that out, it looks cool, can you send in a PR, I think we can replicate the same idea for other mutations.

**Comment by Joebbrien** [comment]
Great, I have re-defined it to send individual error messages, you can test it out. I think this will do on every mutation. Let me know what you think
